src/main/java/it/rate/webapp/controllers/InterestAdminController.java

```diff
@@ -58,15 +58,20 @@ public String editInterest(
 
   @GetMapping("/users")
   @PreAuthorize("hasAnyAuthority(@permissionService.manageCommunity(#interestId))")
-  public String editUsersPage(@PathVariable Long interestId, Model model)
+  public String editUsersPage(@PathVariable Long interestId, Model model, Principal principal)
       throws BadRequestException {
     Optional<Interest> optInterest = interestService.findInterestById(interestId);
     if (optInterest.isEmpty()) {
       throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Interest not found");
     }
-    model.addAttribute("usersByRoles", manageInterestService.getUsersByRole(interestId));
+    if (principal != null) {
+      model.addAttribute(
+          "loggedUser",
+          userService
+              .findByEmail(principal.getName())

```

Comment:
Based on all Matej's comments, I guess this should be getByEmail. :P

Reply:
Done. Also same done for retrieving interest, where failed retrieve scenario is already covered in Preauthorize method